File: 29. Divide Two Integers/Solution.py
# Krishno_MD_KUET_CSE
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        MAX, MIN = 2**31-1,-2**31
        if dividend==MIN and divisor==-1:
            return MAX
        divis = abs(divisor)
        divid = abs(dividend)
        quotent = 0
        while divid>=divis:
            temp,multiple = divis,1
            while divid >= (temp<<1):
                temp <<= 1
                multiple <<= 1
            divid -= temp
            quotent += multiple
        if divisor<0 and dividend<0:
            return quotent
        elif divisor<0 or dividend<0:
            return -quotent
        else:
            return quotent

File: 29. Divide Two Integers/Solution.cpp
// Krishno_MD_KUET_CSE
class Solution {
public:
    int divide(int dividend, int divisor) {
        if(dividend==INT_MIN && divisor==-1) return INT_MAX;
        long divis = abs((long)divisor);
        long divid = abs((long)dividend);
        long quotent = 0;
        while(divid>=divis){
            long temp = divis, multiple = 1;
            while(divid >= (temp<<1)){
                temp <<= 1;
                multiple <<= 1;
            }
            divid -= temp;
            quotent += multiple;
        }
        if(divisor<0 && dividend<0){
            return quotent;
        }else if(divisor<0 || dividend<0){
            return -quotent;
        }else{
            return quotent;
        }
    }
};
